### DISCUSSION TITLE
How to create multiple tours?

### DISCUSSION THREAD

**Discussion by AksanaMaliaus** [hakimio/ngx-ui-tour#176] [Q&A] [answered]
Hi. I need to create multiple tours in my application. Is it possible to do?
I've already tried, but had unexpected behaviour and errors.

**Comment by hakimio** [comment]
You can have one tour at a time. After one finishes you can reconfigure the tour with different steps.

**Reply by AksanaMaliaus** [reply]
So which way is preferable:
1. I create one tour at the root of the project and assign different steps depending on the condition.
2. I create two tours in different Angular components, assign steps to each of them and launch one by one.

And should I explicitly end the previously launched tour with end() method? Or it will be done implicitly if user click on "end" button or "cross" sign?

**Reply by hakimio** [reply, marked as answer]
Option number `2`. The tour service is global to the whole app, you can not have 2 tours configured with different steps at the same time.
You can launch a new tour after the previous one is completed by the user or you explicitly complete it by calling the method. Might be best to simple check the state of the tour with `getState()` before starting the second tour to make sure you are not starting a new tour while the previous one hasn't ended yet.

**Reply by AksanaMaliaus** [reply]
Thank you very much, it works now! But instead of getState() I used end() for current tour before initialization a new one, because there is no getState method in the API.

**Reply by hakimio** [reply]
Sorry, it's called `getStatus()`. See the following example: https://github.com/hakimio/ngx-ui-tour/blob/26b33eda65e3328a155618779748ac689565ee8f/projects/lazy-tour-demo/src/app/components/home/home.component.ts#L19
